### ISSUE TITLE
Training the Swav model with a customized dataset reveals that the size of the .ckpt file is the same for each round of saves.

### ISSUE THREAD

**Issue by ad5014858** [lightly-ai/lightly#1521]
Dear,
I am a beginer,When training with a custom dataset and using the ModelCheckpoint library to save the ckpt generated after each training, it was found that the .ckpt file size was the same for each epoch.

The code is as follows:
```import os
import pytorch_lightning as pl
import torch
import torchvision
from PIL import Image
from torch import nn
from torch.utils.data import Dataset
import torchvision.transforms as transforms
from pytorch_lightning.callbacks import ModelCheckpoint
from lightly.loss import SwaVLoss
from lightly.models.modules import SwaVProjectionHead, SwaVPrototypes
from lightly.models.modules.memory_bank import MemoryBankModule
from lightly.transforms.swav_transform import SwaVTransform
from lightly.data import LightlyDataset


class SwaV(pl.LightningModule):
    def __init__(self):
        super().__init__()
        resnet = torchvision.models.resnet18()
        self.backbone = nn.Sequential(*list(resnet.children())[:-1])
        self.projection_head = SwaVProjectionHead(512, 512, 128)
        self.prototypes = SwaVPrototypes(128, n_prototypes=512)
        self.criterion = SwaVLoss()

    def forward(self, x):
        x = self.backbone(x).flatten(start_dim=1)
        x = self.projection_head(x)
        x = nn.functional.normalize(x, dim=1, p=2)
        p = self.prototypes(x)
        return p

    def training_step(self, batch, batch_idx):
        self.prototypes.normalize()
        views = batch[0]
        multi_crop_features = [self.forward(view.to(self.device)) for view in views]
        high_resolution = multi_crop_features[:2]
        low_resolution = multi_crop_features[2:]
        loss = self.criterion(high_resolution, low_resolution)
        self.log('train_loss',loss)
        return loss

    def configure_optimizers(self):
        optim = torch.optim.Adam(self.parameters(), lr=0.001)
        return optim
        
model = SwaV()
path_to_data = '/root/LightlyAI/lightly/data'
transform = SwaVTransform()
dataset_train_swav = LightlyDataset(input_dir=path_to_data, transform=transform)

dataloader = torch.utils.data.DataLoader(
    dataset_train_swav,
    batch_size=64,
    shuffle=True,
    drop_last=True,
    num_workers=0,
)
checkpoint_callback = ModelCheckpoint(
    monitor='train_loss',
    save_top_k=-1,
    dirpath = './Output/',
    filename = '{epoch}',
    every_n_epochs=1,
    save_weights_only=True,
    
)

accelerator = "gpu" if torch.cuda.is_available() else "cpu"
trainer = pl.Trainer(max_epochs=500, devices=1, accelerator=accelerator,callbacks=checkpoint_callback,)
trainer.fit(model=model, train_dataloaders=dataloader)`
```
![Snipaste_2024-04-09_16-23-23](https://github.com/lightly-ai/lightly/assets/55981213/59f4700e-306c-42a1-95aa-1ae6fa980f47)

I've searched many places and can't find the reason, please tell me why this is, thanks

**Comment by guarin** [comment]
Hi, the checkpoint size should stay pretty much the same size over multiple epochs as the same number of weights are stored in the checkpoint. The size of the checkpoint should only change if you add/remove layers from your model during training. However, the values stored inside the weights should change. You can verify this by loading two checkpoints and comparing the `state_dict` in them.

**Comment by ad5014858** [comment]
> Hi, the checkpoint size should stay pretty much the same size over multiple epochs as the same number of weights are stored in the checkpoint. The size of the checkpoint should only change if you add/remove layers from your model during training. However, the values stored inside the weights should change. You can verify this by loading two checkpoints and comparing the `state_dict` in them.

Thank you very much for your reply, it solved my doubts

**Comment by whiterAutumn** [comment]
Hello, this is my first time encountering unsupervised learning. I would like to know how to generate a custom dataset and whether it accepts multiple categories

**Comment by ad5014858** [comment]
> Hello, this is my first time encountering unsupervised learning. I would like to know how to generate a custom dataset and whether it accepts multiple categories

Hi, you can refer to the 'lightly/examples/pytorch/swav.py' code at lines 44 and 45. My approach is to specify the file path of the dataset (which contains an unsupervised dataset with only images and no labeled data), and then create a dataset class to perform unsupervised training.

My Code about build custom dataset
`path_to_data = 'xx/xx/custom_dataset'
transform = SwavTransform()
dataset_train_swav = LightlyDataset(input_dir = path_to_data,transform = transform)

dataloader = torch.utils.data.DataLoader(
    dataset_train_swav,
    batch_size=128,
    shuffle=True,
    drop_last=True,
    num_workers=0,
)`

**Comment by whiterAutumn** [comment]
Thank you very much for your reply. When I used **lightly/examples/pytorch/swav.py** trained on the VOC2012 dataset, the loss did not decrease significantly，and I only changed the path.
the voc folder formatis:
**VOCdevkit/VOC2012:
--Annotations
--ImageSets
  -- Action
  -- Layout
  -- Main
  -- Segmentation
--JPEGImages
--SegmentationClass
--SegmentationObject**

the loss for 10 epoch:
**epoch: 00, loss: 6.28696
epoch: 01, loss: 6.23867
epoch: 02, loss: 6.23839
epoch: 03, loss: 6.23837
epoch: 04, loss: 6.23836
epoch: 05, loss: 6.23836
epoch: 06, loss: 6.23835
epoch: 07, loss: 6.23835
epoch: 08, loss: 6.23835
epoch: 09, loss: 6.23834**

So I suspect that there is a problem with my data loading format。This also led me to not refer to the format of publicly available datasets. I also test a lower lr and bigger epoch, no improvements.

**Comment by ad5014858** [comment]
> Thank you very much for your reply. When I used **lightly/examples/pytorch/swav.py** trained on the VOC2012 dataset, the loss did not decrease significantly，and I only changed the path. the voc folder formatis: **VOCdevkit/VOC2012: --Annotations --ImageSets -- Action -- Layout -- Main -- Segmentation --JPEGImages --SegmentationClass --SegmentationObject**
> 
> the loss for 10 epoch: **epoch: 00, loss: 6.28696 epoch: 01, loss: 6.23867 epoch: 02, loss: 6.23839 epoch: 03, loss: 6.23837 epoch: 04, loss: 6.23836 epoch: 05, loss: 6.23836 epoch: 06, loss: 6.23835 epoch: 07, loss: 6.23835 epoch: 08, loss: 6.23835 epoch: 09, loss: 6.23834**
> 
> So I suspect that there is a problem with my data loading format。This also led me to not refer to the format of publicly available datasets. I also test a lower lr and bigger epoch, no improvements.
When I use Swav for unsupervised training, the Loss value does not change significantly, but the actual effect is indeed improved. For example, I have used weights of 100/200/300 epochs for unsupervised clustering, and the clustering results are shown in the figure.

![result100](https://github.com/user-attachments/assets/7fc0570e-d5f4-4d6f-8f3c-32eb57e41b16)
![result200](https://github.com/user-attachments/assets/c5378e87-a12a-49ed-bd11-14cc786e0228)
![result300](https://github.com/user-attachments/assets/46ded349-2551-476e-a343-0c90b86c5f2d)

Have you tried training with the official dataset to see if the Loss value still does not show a downward trend but fluctuates instead?

**Comment by whiterAutumn** [comment]
> > Thank you very much for your reply. When I used **lightly/examples/pytorch/swav.py** trained on the VOC2012 dataset, the loss did not decrease significantly，and I only changed the path. the voc folder formatis: **VOCdevkit/VOC2012: --Annotations --ImageSets -- Action -- Layout -- Main -- Segmentation --JPEGImages --SegmentationClass --SegmentationObject**
> > the loss for 10 epoch: **epoch: 00, loss: 6.28696 epoch: 01, loss: 6.23867 epoch: 02, loss: 6.23839 epoch: 03, loss: 6.23837 epoch: 04, loss: 6.23836 epoch: 05, loss: 6.23836 epoch: 06, loss: 6.23835 epoch: 07, loss: 6.23835 epoch: 08, loss: 6.23835 epoch: 09, loss: 6.23834**
> > So I suspect that there is a problem with my data loading format。This also led me to not refer to the format of publicly available datasets. I also test a lower lr and bigger epoch, no improvements.
> 
> When I use Swav for unsupervised training, the Loss value does not change significantly, but the actual effect is indeed improved. For example, I have used weights of 100/200/300 epochs for unsupervised clustering, and the clustering results are shown in the figure.
> 
> ![result100](https://private-user-images.githubusercontent.com/55981213/352350729-7fc0570e-d5f4-4d6f-8f3c-32eb57e41b16.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NjQyNTgsIm5iZiI6MTcyMTk2Mzk1OCwicGF0aCI6Ii81NTk4MTIxMy8zNTIzNTA3MjktN2ZjMDU3MGUtZDVmNC00ZDZmLThmM2MtMzJlYjU3ZTQxYjE2LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDAzMTkxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlOGY3NjY5YmUyNjFjNzRhNWY2YTczZjAyMWFmZDNiNDYzZDgyZmUyNzY4ZmI4ODExYjA0OGM4YmI3NmQ5YzcmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.4HrFu8x6qIEcSvNWIuHWig1w6cIA-g_PcKLL1uDzIY8) ![result200](https://private-user-images.githubusercontent.com/55981213/352351047-c5378e87-a12a-49ed-bd11-14cc786e0228.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NjQyNTgsIm5iZiI6MTcyMTk2Mzk1OCwicGF0aCI6Ii81NTk4MTIxMy8zNTIzNTEwNDctYzUzNzhlODctYTEyYS00OWVkLWJkMTEtMTRjYzc4NmUwMjI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDAzMTkxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlZTMzOGI3MzBiYWM2OTYzYmI5NDIyYTljOWY2OTY3NDBmYmIzNTQyYmYzMjJhYTUyZDNkYmM1Y2MzZTNhZDAmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.cumHDxoVSsc5vgQfj0low1lRbUuiMXWRz-CHaocR47I) ![result300](https://private-user-images.githubusercontent.com/55981213/352351377-46ded349-2551-476e-a343-0c90b86c5f2d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjE5NjQyNTgsIm5iZiI6MTcyMTk2Mzk1OCwicGF0aCI6Ii81NTk4MTIxMy8zNTIzNTEzNzctNDZkZWQzNDktMjU1MS00NzZlLWEzNDMtMGM5MGI4NmM1ZjJkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA3MjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNzI2VDAzMTkxOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWVjM2E5YTZjOGIxYmQ5NDgwNDkwMDc4ZjBiOGI3M2IwMzhhYjRkZjM4ZmU0NGE0Y2FlYzhjMmRiMjZkN2RmNDgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2D7i2PIV72M1eOiIk9L0sWdHijv0dx37_GfW4FDqVmw)
> 
> Have you tried training with the official dataset to see if the Loss value still does not show a downward trend but fluctuates instead?

ok, I'll give it a try in cifar10 first.
